packages/core/src/components/DataTable/DataTable.tsx

```diff
@@ -114,10 +116,7 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     const { dynamicRowHeight, data, filterData, width, height, sortByCacheKey } = this.props;
     const { sortBy, sortDirection } = this.state;
     const dimensionsChanged = width !== prevProps.width || height !== prevProps.height;
-    const sortChanged =
-      sortBy !== prevState.sortBy ||
```

Comment:
we already invoke `this.cache.clearAll();` when updating `sortBy` or `sortByDirection`, so this was doing a double update